### PR TITLE
Full-on switching Breeze development to use uv

### PR DESCRIPTION
**Pull request by potiuk**
Since we are switching to uv, breeze dev environment can now also be managed with uv and if you fully switch to it, the workflow of managing breeze, setting the dev environment and upgrading dependencies is far simpler than airflow's so we should be able to follow the regular uv workflow for it - i.e. use uv to manage dependencies, sync, upgrade them etc. This will be really nice way for some of the maintainers to get to learn uv more and prepare for more standardisation in the future Airflow development (maybe) - when we will link uv.lock and constraints management.

Documentation has been updated in Breeze's README to explain some basic way of using uv for breeze development.

---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/contributing-docs/05_pull_requests.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

File: dev/breeze/README.md [hunk 1]
```diff
@@ -62,6 +63,67 @@ available when the package was installed. Since this file becomes part of the in
...
+```shell
+uv venv
+```
+
+3. Synchronize Breeze dependencies with `uv` to the latest dependencies stored in uv.lock file:
+
+```shell
+uv sync
+```
+
```
Comment (kaxil) [inline dev/breeze/README.md:106]: Worth adding extra example otherwise it won't have pytest installed
```
uv sync --extra devel
```

**Review by kaxil — APPROVED** [review]

File: dev/breeze/README.md [hunk 2]
```diff
@@ -62,6 +63,67 @@ available when the package was installed. Since this file becomes part of the in
...
+and for Breeze. So the instructions below are for setting up the development environment for Breeze
+using `uv`. However we are using only standard python packaging tools, so you can still use `pip` or
+`pipenv` or other build frontends to install Breeze, but we recommend using `uv` as it is the most
+convenient way to install, manage python packages and virtual environments.
+
+Unlike in Airflow, where we manage our own constraints, we use `uv` to manage requirements for Breeze
+and we use `uv` to lock the dependencies. This way we can ensure that the dependencies are always
+up-to-date and that the development environment is always consistent for different people. This is
+why Breeze's `uv.lock` is committed to the repository and is used to install the dependencies by
+default by Breeze.
```
Comment (amoghrajesh) [inline dev/breeze/README.md]:
```suggestion
default by Breeze. Here's how to install breeze with `uv`
```

Comment (amoghrajesh) [inline dev/breeze/README.md]: Do we have to switch the directory to dev?

Comment (amoghrajesh) [reply dev/breeze/README.md]: Shouldnt be needed but it would probably be more "complete" that way

File: dev/breeze/README.md (hunk 1 above)
Comment (potiuk) [reply dev/breeze/README.md:106]: Well. Breeze has no "devel" extra :D

Comment (potiuk) [reply dev/breeze/README.md:106]: All requirements of breeze are "development" requirements simply ...

File: dev/breeze/README.md (hunk 2 above)
Comment (potiuk) [reply dev/breeze/README.md]: This is all about being in "dev/breeze" folder - I will make it celar in the docs,

Comment (potiuk) [reply dev/breeze/README.md]: Oh.. I found out that we have this really nice markdown extension available in Github - to mark notes/important/ warning etc. https://github.com/orgs/community/discussions/16925: 

Screenshot 2024-11-04 at 13 16 16

We should definitely apply it in more of our docs :)

Comment (gopidesupavan) [reply dev/breeze/README.md]: super cool :)

Comment (amoghrajesh) [reply dev/breeze/README.md]: Oh this is nice

**Review by amoghrajesh — APPROVED** [review]

**Review by gopidesupavan — APPROVED** [review]

File: dev/breeze/README.md [hunk 3]
```diff
@@ -62,6 +63,75 @@ available when the package was installed. Since this file becomes part of the in
...
+```
+
+After syncing, the `.venv` directory will contain the virtual environment with all the dependencies
+installed - you can use that environment to develop Breeze - for example with your favourite IDE
+or text editor, you can also use `uv run` to run the scripts in the virtual environment.
+
+For example to run all tests in the virtual environment you can use:
+
+```shell
+    uv run pytest
```
Comment (Lee-W) [inline dev/breeze/README.md]:
```suggestion
uv run pytest
```

**Review by Lee-W — APPROVED** [review]

**Review by phanikumv — APPROVED** [review]

**Review by vincbeck — APPROVED** [review]

**Review by pierrejeambrun — APPROVED** [review]

**Pull request merged**